ci-operator/step-registry/ignition/bastionhost/ignition-bastionhost-commands.sh

```diff
@@ -0,0 +1,394 @@
+reg_brew_user=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.user')
+reg_brew_password=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.password')
+cat > "${patch_file}" << EOF
+proxy:
+  remoteurl: "${reg_brew_url}"
+  username: "${reg_brew_user}"
+  password: "${reg_brew_password}"
+EOF
+/tmp/yq m -x -i "${workdir}/registry_config_file_6002" "${patch_file}"
+
+rm -f "${patch_file}"
+
+for port in "${registry_ports[@]}"; do
+  registry_service_file="${workdir}/registry_service_file_$port"
+  registry_config_file="${workdir}/registry_config_file_$port"
+
+  # adjust system unit content to ignition format
+  #   replace [newline] with '\n', and replace '"' with '\"'
+  #   https://stackoverflow.com/questions/1251999/how-can-i-replace-a-newline-n-using-sed
+  REGISTRY_SERVICE_CONTENT=$(sed ':a;N;$!ba;s/\n/\\n/g' "${registry_service_file}" | sed 's/\"/\\"/g')
+  REGISTRY_CONFIG_CONTENT=$(cat "${registry_config_file}" | base64 -w0)
+
+  registry_ignition_patch=$(mktemp)
+  cat > "${registry_ignition_patch}" << EOF
+{
+  "storage": {
+    "files": [
+      {
+        "path": "/opt/registry-${port}/auth/htpasswd",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_PASSWORD_CONTENT}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry-${port}/certs/domain.crt",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_CRT_CONTENT}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry-${port}/certs/domain.key",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_KEY_CONTENT}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/opt/registry-${port}/config.yaml",
+        "contents": {
+          "source": "data:text/plain;base64,${REGISTRY_CONFIG_CONTENT}"
+        },
+        "mode": 420
+      }
+    ],
+    "directories": [
+      {
+        "path": "/opt/registry-${port}/data",
+        "mode": 493
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "${REGISTRY_SERVICE_CONTENT}",
+        "enabled": true,
+        "name": "poc-registry-${port}.service"
+      }
+    ]
+  }
+}
+EOF
+
+  # patch registry setting to ignition
+  patch_ignition_file "${bastion_ignition_file}" "${registry_ignition_patch}"
+  rm -f "${registry_ignition_patch}"
+done
+
+# update ssh keys
+tmp_keys_json=`mktemp`
+tmp_file=`mktemp`
+echo '[]' > "$tmp_keys_json"
+
+ssh_pub_keys_file="${CLUSTER_PROFILE_DIR}/ssh-publickey"
+readarray -t contents < "${ssh_pub_keys_file}"
+for ssh_key_content in "${contents[@]}"; do
+  jq --arg k "$ssh_key_content" '. += [$k]' < "${tmp_keys_json}" > "${tmp_file}"
+  mv "${tmp_file}" "${tmp_keys_json}"
+done
+
+jq --argjson k "`jq '.| unique' "${tmp_keys_json}"`" '.passwd.users[0].sshAuthorizedKeys = $k' < "${bastion_ignition_file}" > "${tmp_file}"
```

Comment:
a minor comment,  I think one ssh key is enough for Prow CI, like https://github.com/openshift/release/blob/master/ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh#L272-L273

we do support multi-keys in Flexy, it's b/c user have such requirements, e.g. case debugging.

Reply:
@jianlinliu WDTY

Reply:
We can make the code having the capacity, that is not bad things. I am not sure if someday we also had such similar requirement. So I prefer to keep it.